websites/R/RadioMe/presence.ts

```diff
@@ -1,220 +1,131 @@
-                  document.getElementById("bar-ratingCount").textContent
-                } Ratings)`;
-                break;
-            }
-          }
-        } else {
-          //Region
-          presenceData.smallImageKey = "reading";
-          presenceData.smallImageText = (await strings).browse;
-          switch (language) {
-            case "de":
-              presenceData.details = document.querySelector("h1").textContent;
-              presenceData.state = `auf ${host}`;
-              break;
-            case "fr":
-              presenceData.details = document.querySelector("h1").textContent;
-              presenceData.state = `sur ${host}`;
-              break;
-            case "en":
-              presenceData.details = document.querySelector("h1").textContent;
-              presenceData.state = `on ${host}`;
-              break;
-          }
-        }
-      } else if (
-        document.getElementsByClassName("song-name")[0].textContent.length > 0
-      ) {
-        //Player is active
-        if (
-          document.getElementsByClassName(
-            "playbutton-global playbutton-global-playing"
-          ).length > 0
-        ) {
-          //Radio is playing
-          if (
-            !browsingTimestamp ||
-            lastRadio !==
-              document.getElementsByClassName("song-name")[0].textContent
-          )
-            browsingTimestamp = Math.floor(Date.now() / 1000);
-          presenceData.startTimestamp = browsingTimestamp;
-          lastRadio =
-            document.getElementsByClassName("song-name")[0].textContent;
-
-          presenceData.smallImageKey = "play";
-          presenceData.smallImageText = (await strings).play;
-
-          presenceData.details =
-            document.getElementsByClassName("song-name")[0].textContent;
-        } else {
-          //Radio is stopped
-          browsingTimestamp = 0;
-
-          presenceData.smallImageKey = "pause";
-          presenceData.smallImageText = (await strings).pause;
-
-          presenceData.details =
-            document.getElementsByClassName("song-name")[0].textContent;
-        }
-      } else {
-        //Player is inactive
-
-        presence.setActivity();
-        return;
-      }
+      handleUnknown(path);
       break;
   }
+});
+
+async function getStrings() {
+  return presence.getStrings(
+    {
+      play: "general.playing",
+      pause: "general.paused",
+      search: "general.searching",
+      browsing: "general.browsing"
+    },
+    await presence.getSetting<string>("lang").catch(() => "en")
+  );
+}
+
+/**
+ * Handle radio station
+ */
+function handleStation(): void {
+  const station =
+    document.querySelector<HTMLSpanElement>(".song-name")?.textContent;
+
+  let presenceData: PresenceData;
+
+  // Check if the playing icon is shown
+  if (document.querySelector<HTMLDivElement>(".playbutton-global-playing")) {
+    // Radio is playing / buffering
+    timestamp &&= Date.now();
```

Comment:
Nope that’s not what you wanna do
```suggestion
    timestamp ||= Date.now();
```

Reply:
Well, they linked me to the logical AND page. ¯\_(ツ)_/¯
I'm going to change it.